third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -171,8 +168,8 @@ struct ConvertTritonGPUToLLVM
                                                benefit);
     mlir::triton::populateAssertOpToLLVMPattern(typeConverter, patterns,
                                                 targetInfo, benefit);
-    mlir::triton::populateMemoryOpToLLVMPattern(typeConverter, targetInfo,
-                                                patterns, benefit);
+    mlir::triton::NVIDIA::populateMemoryOpToLLVMPatterns(
```

Comment:
Note that `NVIDIA::populateMemoryOpToLLVMPatterns` invokes the general `populateMemoryOpToLLVMPatterns` but assigns a higher priority value to backend optimized conversion. I think the way AMD backend specifies it is a bit cleaner. Not a big issue though